tests/Query/Urgency.test.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import moment from 'moment';
+import { TaskBuilder } from '../TestingTools/TaskBuilder';
+import { Urgency } from '../../src/Urgency';
+import { Priority } from '../../src/Task';
+import { calculateRelativeDate } from '../TestingTools/DateTestHelpers';
+
+window.moment = moment;
+
+function testUrgency(builder: TaskBuilder, expectedScore: number) {
+    const task = builder.build();
+    expect(Urgency.calculate(task)).toBeCloseTo(expectedScore, 5); // 5 digits after decimal point
+}
+
+function testUrgencyOnDate(
+    today: string,
+    builder: TaskBuilder,
+    expectedScore: number,
+) {
+    // TODO Remove this duplication from Task.test.ts
+    const todaySpy = jest
+        .spyOn(Date, 'now')
+        .mockReturnValue(moment(today).valueOf());
+
+    testUrgency(builder, expectedScore);
+
+    todaySpy.mockClear();
+}
+
+/**
+ * Return a TaskBuilder to create a Task with Low Priority.
+ *
+ * Priority Low adds zero to the score, which means the
+ * test for other parts of the priority calculation becomes clearer.
+ */
+function lowPriorityBuilder() {
+    return new TaskBuilder().priority(Priority.Low);
+}
+
+function testUrgencyForDueDate(daysToDate: number, expectedScore: number) {
```

Comment:
It's a bit of a wall of helper functions at the top.
I'm thinking I'll move the functions for testing with the various date fields down to just before they are used, to reduce the scope, and see if it might make things clearer.

Reply:
I did this, and it helped.